### PR TITLE
Bump OT to 0.30.0

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java [hunk 1]
```diff
@@ -59,7 +57,8 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                 spanBuilder.asChildOf(extractedSpanContext);
             }
 
-            Span span = spanBuilder.start();
+            Span span = spanBuilder.startManual();
+            tracer.makeActive(span);
```
Comment (pavolloffay) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: This is ugly and the reason for this is that in async requests the thread in response filter is different and active span cannot be deactivated/finished: https://github.com/opentracing/opentracing-java/blob/master/opentracing-util/src/main/java/io/opentracing/util/ThreadLocalActiveSpan.java#L50.  

The most ideal solution would be to instrument `ExecutorService`/or something like `onAsync` but I could not find a way how to do it in jax-rs.

Other options:
1. do not activate at all (not wanted)
2. create "manual" span and activate explicitly. 
3. activate only for sync requests. (this is Brave doing https://github.com/pavolloffay/brave/blob/629a7da8d8e650124918b0e613b68bedd39cef34/instrumentation/jaxrs2/src/main/java/brave/jaxrs2/TracingContainerFilter.java#L74-L74)

cc @jpkrohling @objectiser

Comment (malafeev) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: `tracer.makeActive(span);` - will make span active forever? (if thread is taken from thread pool and returned back when execution is finished)

Comment (pavolloffay) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: @malafeev exactly, I think 3. is the most consistent solution. If we cannot deactivate span then do not activate. I'm wondering what other people think about this.

Comment (objectiser) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: Have you tried using a continuation?

Comment (pavolloffay) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: What do you mean e.g. could you outline possible solution? What should be done in request and response filters.

Comment (objectiser) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: One possible approach would be to replace in the request filter `requestContext.setProperty(SPAN_PROP_ID, new SpanWrapper(span));` by a request context property that stores the `Continuation` returned from `span.capture()` - and on a following line do `span.deactivate()`.

Then in the response filter retrieve the `Continuation` and `activate` it to retrieve the active span which can then be `deactivate`d at the end of the method.

Comment (malafeev) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: @objectiser if span is deactivated in request filter then during processing of request (before response filter) span will not be picked up as parent span. In such case there is no sense to activate/deactivate span. 
For example:
1. Send request
2. Request Endpoint send another REST request. Here we should get active span from request filter to use as parent for new REST request.
3. Send response

Comment (pavolloffay) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: > One possible approach would be to replace in the request filter requestContext.setProperty(SPAN_PROP_ID, new SpanWrapper(span)); by a request context property that stores the Continuation returned from span.capture() - and on a following line do span.deactivate().

> Then in the response filter retrieve the Continuation and activate it to retrieve the active span which can then be deactivated at the end of the method.

As @malafeev said this does not make sense, we could simply use `Span` and just finish it.

Comment (objectiser) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: Yeah was forgetting about the child spans :)

Comment (objectiser) [inline opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java]: @pavolloffay Would be worth creating an issue on opentracing-java to see if a general solution can be found.

**Comment by pavolloffay** [comment]
PR updated.

Based on discussion with @jpkrohling, span is activated only in sync requests, because we should make sure that the span can be properly deactivated.

**Comment by pavolloffay** [comment]
@jpkrohling @objectiser could you please re-review? Or is it good to go?

File: README.md [hunk 2]
```diff
@@ -4,49 +4,100 @@
...
+  .request()
+  .property(TracingProperties.CHILD_OF, parentSpanContext) // optional, by default new parent is inferred from span source
+  .property(TracingProperties.TRACING_DISABLED, false) // optional, by default everything is traced
+  .get();
+```
+
+### Async
+Async requests are executed in a different thread than when the client has been invoked, therefore
+spans representing client requests are not connected to appropriate parent. To fix this JAX-RS client
+has to use OpenTracing-aware `ExecutorService`.
```
Comment (pavolloffay) [inline README.md]: +1 will add

Comment (pavolloffay) [inline README.md]: Link added, actually that dependency is pulled by default so users would find it.

**Comment by pavolloffay** [comment]
@jpkrohling @objectiser thanks

**Pull request merged**